build.gradle

```diff
@@ -11,15 +14,35 @@ ext {
     opentelemetryProjects = subprojects - project(":opentelemetry-bom")
 }
 
+release {
+    defaultVersionStrategy = Strategies.getSNAPSHOT()
+}
+
+nebulaRelease {
+    addReleaseBranchPattern(/v\d+\.\d+\.x/)
+}
+
+def releaseTask = tasks.named("release")
+releaseTask.configure {
+    mustRunAfter("snapshotSetup", "finalSetup")
+}
+
 subprojects {
     group = "io.opentelemetry"
-    version = "0.10.0-SNAPSHOT" // CURRENT_OPEN_TELEMETRY_VERSION
 
     plugins.withId("maven-publish") {
         // Always include the artifactory/bintray plugins to do the deployment.
         pluginManager.apply "com.jfrog.artifactory"
         pluginManager.apply "com.jfrog.bintray"
 
+        releaseTask.configure {
+            if (version.toString().endsWith('-SNAPSHOT')) {
+                finalizedBy(tasks.named('artifactoryPublish'))
```

Comment:
This will run them even if `releaseTask` fails, are you sure?

Reply:
`releaseTask` is just a placeholder, so it can't fail. I think it's ok - it's the simplest way to have publishing guaranteed to happen after the dependencies of `releaseTask` instead of wiring dependencies if needed.